ODD number of argument.

Hi All,

We are having below error in solr-operator.

DPANIC	controller	odd number of arguments passed as key-value pairs for logging	{"ignored key": ""}
github.com/go-logr/zapr.handleFields
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:106
github.com/go-logr/zapr.(*infoLogger).Info
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:70
github.com/bloomberg/solr-operator/controllers/util.CreateCollection
	/workspace/controllers/util/collection_util.go:46
github.com/bloomberg/solr-operator/controllers.reconcileSolrCollection
	/workspace/controllers/solrcollection_controller.go:172
github.com/bloomberg/solr-operator/controllers.(*SolrCollectionReconciler).Reconcile
	/workspace/controllers/solrcollection_controller.go:69
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:88
E0309 01:40:20.090053       1 runtime.go:76] Observed a panic: odd number of arguments passed as key-value pairs for logging
goroutine 359 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x13a1880, 0xc000647740)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/runtime/runtime.go:48 +0x82
panic(0x13a1880, 0xc000647740)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0000efb80, 0xc0003aa780, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.9.1/zapcore/entry.go:229 +0x546
go.uber.org/zap.(*Logger).DPanic(0xc00007b2c0, 0x15fa803, 0x3d, 0xc0003aa780, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.9.1/logger.go:215 +0x7f
github.com/go-logr/zapr.handleFields(0xc00007b2c0, 0xc000647700, 0x1, 0x1, 0x0, 0x0, 0x0, 0x7f8904d9e008, 0x0, 0x0)
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:106 +0x5ce
github.com/go-logr/zapr.(*infoLogger).Info(0xc0002ef2e8, 0x15fc4a1, 0x3e, 0xc000647700, 0x1, 0x1)
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:70 +0xb1
github.com/bloomberg/solr-operator/controllers/util.CreateCollection(0xc0003d30c0, 0x7, 0xc000491920, 0x12, 0x1, 0x1, 0x0, 0x1, 0x0, 0x0, ...)
	/workspace/controllers/util/collection_util.go:46 +0x5c4
github.com/bloomberg/solr-operator/controllers.reconcileSolrCollection(0xc000319740, 0xc000330540, 0x1, 0x1, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/workspace/controllers/solrcollection_controller.go:172 +0x2b8
github.com/bloomberg/solr-operator/controllers.(*SolrCollectionReconciler).Reconcile(0xc000319740, 0xc000056ce9, 0x7, 0xc000491920, 0x12, 0xc00047fcd8, 0xc000312cf0, 0xc0000f23f8, 0x1795120)
	/workspace/controllers/solrcollection_controller.go:69 +0x375
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000336b40, 0x1458380, 0xc0003adf40, 0xc00040cd00)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000336b40, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc000336b40)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000647640)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000647640, 0x3b9aca00, 0x0, 0x1, 0xc000316180)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000647640, 0x3b9aca00, 0xc000316180)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:193 +0x328
panic: odd number of arguments passed as key-value pairs for logging [recovered]
	panic: odd number of arguments passed as key-value pairs for logging
goroutine 359 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/runtime/runtime.go:55 +0x105
panic(0x13a1880, 0xc000647740)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0000efb80, 0xc0003aa780, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.9.1/zapcore/entry.go:229 +0x546
go.uber.org/zap.(*Logger).DPanic(0xc00007b2c0, 0x15fa803, 0x3d, 0xc0003aa780, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.9.1/logger.go:215 +0x7f
github.com/go-logr/zapr.handleFields(0xc00007b2c0, 0xc000647700, 0x1, 0x1, 0x0, 0x0, 0x0, 0x7f8904d9e008, 0x0, 0x0)
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:106 +0x5ce
github.com/go-logr/zapr.(*infoLogger).Info(0xc0002ef2e8, 0x15fc4a1, 0x3e, 0xc000647700, 0x1, 0x1)
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:70 +0xb1
github.com/bloomberg/solr-operator/controllers/util.CreateCollection(0xc0003d30c0, 0x7, 0xc000491920, 0x12, 0x1, 0x1, 0x0, 0x1, 0x0, 0x0, ...)
	/workspace/controllers/util/collection_util.go:46 +0x5c4
github.com/bloomberg/solr-operator/controllers.reconcileSolrCollection(0xc000319740, 0xc000330540, 0x1, 0x1, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/workspace/controllers/solrcollection_controller.go:172 +0x2b8
github.com/bloomberg/solr-operator/controllers.(*SolrCollectionReconciler).Reconcile(0xc000319740, 0xc000056ce9, 0x7, 0xc000491920, 0x12, 0xc00047fcd8, 0xc000312cf0, 0xc0000f23f8, 0x1795120)
	/workspace/controllers/solrcollection_controller.go:69 +0x375
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000336b40, 0x1458380, 0xc0003adf40, 0xc00040cd00)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000336b40, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc000336b40)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000647640)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000647640, 0x3b9aca00, 0x0, 0x1, 0xc000316180)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000647640, 0x3b9aca00, 0xc000316180)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:193 +0x328

--- a/controllers/util/collection_util.go
+++ b/controllers/util/collection_util.go
@@ -43,7 +43,7 @@ func CreateCollection(cloud string, collection string, numShards int64, replicat
 		queryParams.Add("router.name", routerName)
 		queryParams.Add("numShards", numShardsParameter)
 	} else {
-		log.Info("router.name must be either compositeId or implicit. Provided: ", routerName)
+		log.Info("router.name must be either compositeId or implicit. Provided: ", "router.name", routerName)
 	}
 
 	resp := &SolrAsyncResponse{}

